doc/decisions/1_in_discussion/spec_expressiveness.md

```diff
@@ -2,22 +2,155 @@
+This REST application has to use multiple services.
+
+If we have `database`, `broker` and `mailer` services we could use the specification:
+
+```ni
+[service/database/port]
+meta:/require = true
+meta:/default = 8080
+meta:/description = "The port of the database service."
+
+[service/database/url]
+meta:/require = true
+meta:/description = "The url of the database service."
+
+[service/broker/port]
+meta:/require = true
+meta:/default = 8080
+meta:/description = "The port of the broker service."
+
+[service/broker/url]
+meta:/require = true
+meta:/description = "The url of the broker service."
+
+[service/mailer/port]
+meta:/require = true
+meta:/default = 8080
+meta:/description = "The port of the mailer service."
+
+[service/mailer/url]
+meta:/require = true
+meta:/description = "The url of the mailer service."
+```
+
+This is already very repetitive and error prone. 
+But writing the specification becomes completely impossible when the set of services is not known in advance. 
+To solve this we need some way to allow for dynamic/unknown parts in the keyname.
+
+`Globbing` expressions is the solution to this problem. 
+By providing certain wildcard characters e.g. `_` or `#` one can match multiple keys.
+This removes the need to explicitly define specification for keys in our example for every service. 
+
+The thing to be aware of is to define configuration keys which are necessary for every service.
+In this example we use `port` and `url`.
+
+The parent key for our specification is `/sw/org/app-name/#0/development`.
+
+```ni
+[service/_/port]
+meta:/require = true
+meta:/default = 8080
+meta:/description = "The port of the service."
+
+[service/_/url]
+meta:/require = true
+meta:/description = "The url of the service."
+```
+The `_` would match every `service` we are using in our application e.g. `database`, `message broker`, etc.
+
+But the current implementation has some undefined behaivour when using `globbing`.
+
+This specification uses `_` as wildcard character.
+Unfortunately `_` has a shortcoming, as we do not know the key names in advance.
+This prevents us from creating those keys in namespace they do not exist or copying metadata in advance.
+
+But besides `_` we also have `#` as wildcard character.
+This wildcard character is used with arrays.
+A fixed array size can be defined in the specification, which overcomes the shortcoming of `_`.
+
+The problem we face with those wildcard characters is overlapping specifications.
+A example for this is if we used `[service/#/port]` and `[service/shtg/port]` or `[service/_/port]` in the same specification.
+We don't know what metadata should be copied to other namespaces in this case.
+Both `[service/#/port]` and `[service/_/port]` existing produces conflicting specification.
+
+For example `[service/#/port]` having as type `meta:/type = string`.
+`[service/shtg/port]` having `meta:/type = unsigned_short`.
+The spec plugin can not know which of those specification to use.
```

Comment:
Again: this is already solution.

Btw. Of course some behavior could be defined, so the spec plugin can know what to use.

Reply:
> Btw. Of course some behavior could be defined, so the spec plugin can know what to use.

So you are saying that this is not a problem at all? Having both `[service/#/port]` and `[service/shtg/port]` with different `meta:/type` can be configured the way it works?